src/conversions/chrono.rs

```diff
@@ -0,0 +1,848 @@
+            sec in 0u32..=60u32,
+            micro in 0u32..=2_000_000u32
+        ) {
+            // Test roundtrip convertion rust->python->rust for naive times.
+            // Python time has a resolution of microseconds, so we only test
+            // NaiveTimes with microseconds resolution, even if NaiveTime has nanosecond
+            // resolution.
+            Python::with_gil(|py| {
+                // We use to `from_hms_micro_opt` constructor so that we only test valid `NaiveTime`s.
+                // This is to skip the test if we are creating an invalid time
+                if let Some(time) = NaiveTime::from_hms_micro_opt(hour, min, sec, micro) {
+                    let pytime = time.into_py(py);
+                    let roundtripped: NaiveTime = pytime.extract(py).expect("Round trip");
+                    assert_eq!(time, roundtripped);
+                }
+            })
+        }
+
+        #[test]
+        fn test_utc_datetime_roundtrip(
+            year in 1i32..=9999i32,
+            month in 1u32..=12u32,
+            day in 1u32..=31u32,
+            hour in 0u32..=24u32,
+            min in 0u32..=60u32,
+            sec in 0u32..=60u32,
+            micro in 0u32..=2_000_000u32
+        ) {
+            Python::with_gil(|py| {
+                let date_opt = NaiveDate::from_ymd_opt(year, month, day);
+                let time_opt = NaiveTime::from_hms_micro_opt(hour, min, sec, micro);
+                if let (Some(date), Some(time)) = (date_opt, time_opt) {
+                    let dt: DateTime<Utc> = DateTime::from_utc(NaiveDateTime::new(date, time), Utc);
+                    let pydt = dt.into_py(py);
+                    let roundtripped: DateTime<Utc> = pydt.extract(py).expect("Round trip");
+                    assert_eq!(dt, roundtripped);
+                }
+            })
+        }
+
+        #[test]
+        fn test_fixedoffset_datetime_roundtrip(
+            year in 1i32..=9999i32,
+            month in 1u32..=12u32,
+            day in 1u32..=31u32,
+            hour in 0u32..=24u32,
+            min in 0u32..=60u32,
+            sec in 0u32..=60u32,
+            micro in 0u32..=2_000_000u32,
+            offset_secs in -86_400i32..=86_400i32
+        ) {
+            Python::with_gil(|py| {
+                let date_opt = NaiveDate::from_ymd_opt(year, month, day);
+                let time_opt = NaiveTime::from_hms_micro_opt(hour, min, sec, micro);
+                let offset = FixedOffset::east(offset_secs);
+                if let (Some(date), Some(time)) = (date_opt, time_opt) {
+                    let dt: DateTime<FixedOffset> = DateTime::from_utc(NaiveDateTime::new(date, time), offset);
+                    let pydt = dt.into_py(py);
+                    let roundtripped: DateTime<FixedOffset> = pydt.extract(py).expect("Round trip");
+                    assert_eq!(dt, roundtripped);
+                }
+            })
+        }
+    }
+}
+
+#[cfg(test)]
+mod test_chrono {
```

Comment:
Nit: all test modules called `tests` please.

```suggestion
mod tests {
```